### ISSUE TITLE
Narrator reads the text inside Nav component twice

### ISSUE THREAD

**Issue by lijunle** [OfficeDev/office-ui-fabric-react#7052]
### Environment Information 
 
- __Package version(s)__: 6.99.1 (newest)
- __Browser and OS versions__: Windows 10 + Edge + Narrator

### Please provide a reproduction of the bug in a codepen:
 
1. Open this page, https://developer.microsoft.com/en-us/fabric#/components/nav
2. Launch Narrator.
3. Tab to the links in the Nav component.
4. The link buttons read twice.

#### Expected behavior:
 
The text should only read once.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? **Yes**
 
Requested priority: **High**
 
Products/sites affected: **SharePoint**
 
Maybe related: #4869

**Comment by lijunle** [comment]
This is happening because `Nav` component pass the `link.name` to [`title` props](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx#L128) as same as text. Narrator is reading the link text first, then the link title. 

In a `title` attribute best practice, I think this is a wrong usage on the `title` attribute. I could suggest to remove the `title` props here. More ideas?

**Comment by kenotron** [comment]
I see this happening as well (though the issue is exacerbated by the fact that it also reads the GROUP which is potentially yet another duplication).

@betrue-final-final - can we simply get rid of title attrib unless a title prop has been passed in?

**Comment by betrue-final-final** [comment]
Hmmm, well I'm more of an end user UX specialist and am a little fuzzy on the implementation details. @cschlechty would know more.

**Comment by cschlechty** [comment]
Title is needed in areas where there's text truncation. The reading of tooltips is a user preference in nearly all ATs.

**Comment by ipip2005** [comment]
@cschlechty Shouldn't `aria-label` always be read instead of the truncated text?
I there is a text truncation, add `title` would read both the truncated text and the full text which doesn't make sense to me.
I suppose the proper behavior should be always assign text to `aria-label` which is supposed to be the name of the control instead of using `title` or `aria-description`

**Comment by cschlechty** [comment]
You're forgetting that we need tooltips for non-screen reader users as well (mousing, when zoomed, etc). There's a purpose to having tooltips available in the platform. 

The screen reader doesn't read with truncation. It always reads the visible text, and **_might_** read the additional text, like title, describedby, etc.

**Comment by kenotron** [comment]
@cschlechty, Is this a narrator bug where it is reading both in your opinion?

**Comment by cschlechty** [comment]
@kenotron we can ask them about this "feature". :)

**Comment by ipip2005** [comment]
@cschlechty 
For tooltip, can we use the Tooltip component from OUFR instead? Which will not be readout by ATs. I believe we are doing this in sp-client for some components because it gives us flexibility to choose whether link the tooltip text to the element. (e.g. Point aria-describedby to tooltip element)

And when aria-label is specified, it should read the aria-label instead of visible text inside the element. No?

Is there going to be a consistent experience for tooltip across the repo? I would propose using Tooltip component for showing a tooltip while hover/focus and only assign the full text to `aria-label` so ATs are supposed to only read the full text once.

IMHO, it makes sense for Narrator to read both text and title because they have different design purpose. Text is what's inside the element and is supposed to be the element's name while aria-label is not present. Title is usually some instructions, hints, or sometimes can be full text of what is being truncated.

The friction here is `title` is used by browser for two purpose combined:
1. Show as tooltip when hover/focus
2. Read by Narrator
Due to the difference among ATs implementation, it's usually hard to control how title is going to be read.

**Comment by cschlechty** [comment]
@ipip2005 you're trading accessibility issues. Having OUIFR tooltips not be read is a bit of a hack. You're not properly associating the elements, and presumably requiring consumers to jam info into labels that's otherwise already visually present. If you're doing this, you should probably be marking the tooltips hidden.

Let's start a thread with the Narrator/Edge folks. Also, if there's a filed bug with MAS reasoning, can you email a link? I don't think this actually violates a rule, it just sounds wrong.

**Comment by gingi** [comment]
I don't think this is a Narrator/Edge issue. The problem is that the `<span class="ms-Button-ScreenReaderText>` element is not hidden but rather made invisible with CSS positioning, so Narrator doesn't interpret it as hidden. I think the solution is to hide the visual text (the `<div class="ms-Nav-linkText">` element) using `aria-hidden="true"`.

**Comment by cschlechty** [comment]
Ok, yes. I don't get why there's hidden text there. It doesn't appear to be needed. Removing it, and aria-describedby, should fix the double reading.

![image](https://user-images.githubusercontent.com/20362383/49028490-61cbe000-f157-11e8-92da-18e967e171fe.png)
